File: day-2.js
const fs = require('fs')
const path = require("path")

var t = 0
const directory = __dirname
const filePath = path.join(directory, "test.txt")

const fileContent = fs.readFileSync(filePath, "utf-8")
const linesArray = fileContent.split(/\r?\n/)

linesArray.forEach((game) => {
    const [game_id, subsets] = game.split(": ")
    var subset = []
    subset = subsets.split("; ")

    var valid_game = true
    for (var i = 0; i < subset.length; i++){
        var set = []
        set = subset[i].split(", ")
        var initial = { "red": 0, "green": 0, "blue": 0 }

        for (var ball = 0; ball < set.length; ball++){
            var [count, color] = set[ball].split(" ")
            initial[color] = parseInt(count)
            if (initial["red"] > 12 || initial["green"] > 13 || initial["blue"] > 14){
                valid_game = false
            }
        }
        
    }
    if (valid_game) {
        t += parseInt(game_id.split(" ")[1])
    }
})
console.log(t)

File: day-2.py

t = 0
with open("test.txt", mode="r") as file:
    for line in file:
        game_id, subsets = line.split(": ")
        subset = subsets.strip().split("; ")
        print(subset)
        print("=="*50)
        
        valid_game = True
        for set in subset:
            for ball in set.split(", "):
                initial = {"red": 0, "green": 0, "blue":0}
                count, color = ball.split()
                initial[color] = int(count)
                if initial["red"] > 12 or initial["green"] > 13 or initial["blue"] > 14:
                    valid_game = False
        if valid_game:
            t += int(game_id.split()[1])
    print(t)


